core/src/main/java/org/apache/calcite/sql/fun/SqlNewOperator.java

```diff
@@ -49,6 +50,14 @@ public SqlNode rewriteCall(SqlValidator validator, SqlCall call) {
   public boolean requiresDecimalExpansion() {
     return false;
   }
+
+  @Override public boolean validRexOperands(int count, Litmus litmus) {
+    if (count <= 0) {
+      return litmus.fail("wrong operand count {} for {}, Need at least one RexOperand!",
+          count, this);
+    }
+    return litmus.succeed();
+  }
```

Comment:
Here is the java doc of `SqlNewOperator`:
>SqlNewOperator represents an SQL <code>new specification</code> such as
<code>NEW UDT(1, 2)</code>. When used in an SqlCall, SqlNewOperator takes a
single operand, which is an invocation of the constructor method; but when
used in a RexCall, the operands are the initial values to be used for the new
instance.

The `validRexOperands` method is only invoked in `RexCall` constructor, so i think overriding is the right way to go. But why we must require the count bigger than 0, how about there is a user defined constructor without any operands ?

Reply:
hi @danny0405  Thank you very much,Let me know more about it.I address commits for this pr.